week2/project/ecommerce/src/pages/productsList/ProductsList.jsx

```diff
@@ -0,0 +1,52 @@
+import Categories from './categories/Categories'
+import Products from './products/Products'
+import getCategories from '../../data/categories'
+import getProducts from '../../data/products'
+import { useState, useEffect } from 'react'
+import ErrorMessage from '../../components/ErrorMessage'
+
+function ProductsList() {
+    const [categories, setCategories] = useState([]);
+    const [selectedCategory, setSelectedCategory] = useState('');
+    const [products, setProducts] = useState([]);
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState({ isError: false, message: '' })
+
+    useEffect(() => {
+        (async () => {
+            const data = await getCategories();
+            setCategories(data);
+        })()
+    }, [])
```

Comment:
awesome.
you used the IIF immediate invoked function and also made a special function to get the categories for you.
super 🚀

but what if some error happened ?
✏️need to modify✏️

Reply:
added error catching